### PR TITLE
python3-tensorrt: add recipe

### PR DESCRIPTION
**Pull request by madisongh**
Signed-off-by: Matt Madison <matt@madison.systems>

### CODE REVIEW

File: external/meta-python/recipes-devtools/python3-tensorrt/python3-tensorrt_7.1.3-1.bb
Comment (ichergui) [inline external/meta-python/recipes-devtools/python3-tensorrt/python3-tensorrt_7.1.3-1.bb:12]: I think you can remove the `branch=` because will clone the repo by using the `master` branch by default
In this case, no need to define `SRCBRANCH`

File: external/meta-python/recipes-devtools/python3-tensorrt/python3-tensorrt_7.1.3-1.bb
Comment (ichergui) [inline external/meta-python/recipes-devtools/python3-tensorrt/python3-tensorrt_7.1.3-1.bb:49]: I think that `distutils3_do_compile` is not needed because it will be done by default when inheriting `distutils3`

File: external/meta-python/recipes-devtools/python3-tensorrt/python3-tensorrt_7.1.3-1.bb
Comment (ichergui) [inline external/meta-python/recipes-devtools/python3-tensorrt/python3-tensorrt_7.1.3-1.bb:53]: The same as above

**Review by ichergui — COMMENTED** [review]
I'm doing a quick test on taking account my comments

File: external/meta-python/recipes-devtools/python3-tensorrt/python3-tensorrt_7.1.3-1.bb
Comment (ichergui) [reply external/meta-python/recipes-devtools/python3-tensorrt/python3-tensorrt_7.1.3-1.bb:53]: This one is needed

**Review by ichergui — CHANGES_REQUESTED** [review]
I tested the recipe `do_compile_append` is not needed

File: external/meta-python/recipes-devtools/python3-tensorrt/python3-tensorrt_7.1.3-1.bb
Comment (madisongh) [reply external/meta-python/recipes-devtools/python3-tensorrt/python3-tensorrt_7.1.3-1.bb:12]: I know it's the default, but these are there to make it easier to override the repo location and branch via a bbappend or local.conf setting.

File: external/meta-python/recipes-devtools/python3-tensorrt/python3-tensorrt_7.1.3-1.bb
Comment (madisongh) [reply external/meta-python/recipes-devtools/python3-tensorrt/python3-tensorrt_7.1.3-1.bb:51]: I've reworked each of the task definitions to explicitly call on the relevant functions from the inherited classes in the right order.  I have left in the distutils3_do_compile function because it does actually do something.  Yes, distutils3_do_install will also execute the compile tasks, but it does so under pseudo context, which could potentially cause some issues.

File: external/meta-python/recipes-devtools/python3-tensorrt/python3-tensorrt_7.1.3-1.bb
Comment (ichergui) [reply external/meta-python/recipes-devtools/python3-tensorrt/python3-tensorrt_7.1.3-1.bb:12]: Agree

**Review by ichergui — APPROVED** [review]
Looks good to me.

**Comment by madisongh** [comment]
I'll wait for some feedback from @tirankas before merging.

**Comment by tirankas** [comment]
Hi Matt,
I check before the latest change (since two hours ago) and it worked fine.
I will take the last version now and will verify again

**Comment by tirankas** [comment]
Hi Matt,
With the latest changes (picking the two last commits), I get this:
>>> import tensorrt as trt
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python3.8/site-packages/tensorrt/__init__.py", line 35, in <module>
    from .tensorrt import *
ModuleNotFoundError: No module named 'tensorrt.tensorrt'

It worked fine without the last two commits:
>>> import tensorrt as trt
>>> trt.Logger()
<tensorrt.tensorrt.Logger object at 0x7f9d0c6af0>

Not sure what changes broke this...

Thanks



Thanks

**Comment by madisongh** [comment]
Hmm.  OK, thanks, I'll check it out

**Comment by madisongh** [comment]
@tirankas  not sure what happened in your build, but it's working for me.  Can you try a `bitbake -c clean -c cleansstate python3-tensorrt` then build again, just to make sure it's not due to some stale files left around from the earlier build?

**Comment by tirankas** [comment]
Sure, I will check again although I did the same procedure twice ( a to b and then b to a, got consistent results).

**Comment by tirankas** [comment]
Checked again and it is working.
I found one line deleted (within the do_compile function) which lead to the issue, everything works fine now.

Thanks again

**Comment by ichergui** [comment]
Hey @tirankas 
Thanks for the updates. I'm happy that works fine to you.

**Pull request merged**
